src/sentry/lang/native/appconnect.py

```diff
@@ -156,10 +156,17 @@ def from_json(cls, data: Dict[str, Any]) -> "AppStoreConnectConfig":
         return cls(**data)
 
     @classmethod
-    def all_for_project(cls, project: Project) -> "List[AppStoreConnectConfig]":
+    def from_option(cls, raw_sources_option: str) -> "List[AppStoreConnectConfig]":
+        """Creates a list of new instances from a raw serialized JSON list of symbol sources,
+        typically fetched directly from a project's ``sentry:symbol_sources`` option. Useful when
+        App Store Connect symbol sources need to be updated across multiple projects.
+
+        :raises InvalidConfigError: if the data does not contain a valid App Store Connect
+           symbol source configuration.
+        """
+
         sources = []
-        raw = project.get_option(SYMBOL_SOURCES_PROP_NAME, default="[]")
-        all_sources = json.loads(raw)
+        all_sources = json.loads(raw_sources_option)
```

Comment:
We captured an error because the sources config is completely empty for some projects. Might be a good idea to just early return when `raw_sources_option` is empty.